extensions-core/mysql-metadata-storage/src/main/java/io/druid/metadata/storage/mysql/MySQLConnector.java

```diff
@@ -57,6 +63,61 @@ public MySQLConnector(Supplier<MetadataStorageConnectorConfig> config, Supplier<
     // so we need to help JDBC find the driver
     datasource.setDriverClassLoader(getClass().getClassLoader());
     datasource.setDriverClassName("com.mysql.jdbc.Driver");
+    datasource.addConnectionProperty("useSSL", String.valueOf(connectorConfig.isUseSSL()));
+    if (connectorConfig.isUseSSL()) {
+      log.info("SSL is enabled on this MySQL connection. ");
+
+      datasource.addConnectionProperty("verifyServerCertificate", String.valueOf(connectorConfig.isVerifyServerCertificate()));
+      if (connectorConfig.isVerifyServerCertificate()) {
+        log.info("Server certificate verification is enabled. ");
+        if (connectorConfig.getTrustCertificateKeyStoreUrl() != null) {
```

Comment:
Since this is required when `verifyServerCertificate` is set to true, what happens when user does not set this and it is `null`. Will there be some explicit exception/warning in the logs which would make it obvious for user what's the problem ? otherwise it would be better to explicitly check and throw exception.

Reply:
The user does receive a generic error message that the TrustManager was not configured correctly, but it does not tell them the reason was because it was null. I'll add a check for this and update the PR.

Reply:
According to the Mysql docs, the user can leave these properties as null, in which case the default values would be used. The password can also be set to null. In that case, I'm not sure if we need to enforce a null check here. I have clarified this in the documentation. Please review.